src/esm_runscripts/filelists.py

```diff
@@ -774,65 +775,51 @@ def replace_year_placeholder(config):
 def log_used_files(config):
     if config["general"]["verbose"]:
         print("\n::: Logging used files", flush=True)
+
     filetypes = config["general"]["relevant_filetypes"]
     expid = config["general"]["expid"]
     it_coupled_model_name = config["general"]["iterative_coupled_model"]
     datestamp = config["general"]["run_datestamp"]
+    thisrun_log_dir = config["general"]["thisrun_log_dir"]
+    flist_file = (
+        f"{thisrun_log_dir}/{expid}_{it_coupled_model_name}filelist_{datestamp}.yaml"
+    )
+    all_files = {}
+
     for model in config["general"]["valid_model_names"] + ["general"]:
-        thisrun_config_dir = config[model]["thisrun_config_dir"]
-        # this file contains the files used in the experiment
-        flist_file = (
-            f"{thisrun_config_dir}/"
-            f"{expid}_{it_coupled_model_name}filelist_{datestamp}"
-        )
+        for filetype in filetypes:
+            model_config = config[model]
+            model_files = {}
+
+            for file in model_config.get(f"{filetype}_sources", []):
+                try:
+                    checksum = hashlib.md5(
```

Comment:
One nice check to have in mind: does this give the same answer as shell md5 tool? (It should, or I will be very confused)

Reply:
I tried it for a file and it yields the same result (at least on my Mac for that file)